File: internal/testfixture/testdata/fixture-polyglot/pkg/validate/validate.go
// Package validate provides input validation for the fixture repo.
package validate

import "strings"

// Validate reports whether s is a non-empty, trimmed string.
//
// It exists in this fixture specifically to collide in name with
// src/validate.ts's exported Validate: the ambiguous symbol that
// edge-resolution tests use to prove name-based matching stays intra-language
// and does not merge the Go and TypeScript hits together.
func Validate(s string) bool {
	return strings.TrimSpace(s) != ""
}

File: internal/testfixture/testdata/fixture-polyglot/src/validate.ts
/**
 * Validate reports whether the given value is a non-empty, trimmed string.
 *
 * This is the fixture's ambiguous symbol: a TypeScript export named Validate
 * that collides in name with pkg/validate/validate.go's Go export Validate,
 * proving that name-based edge resolution must not merge cross-language
 * hits.
 */
export function Validate(value: string): boolean {
  return value.trim().length > 0;
}
